python/pyspark/sql/decorators.py

```diff
@@ -0,0 +1,25 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.sql.types import StringType
+from pyspark.sql.functions import udf as _udf
+
+
+def udf(returnType=StringType()):
```

Comment:
Regarding name clashes I would like to keep it as a separated module and let user decide if prefer to import from `sql.functions` or `sql.decorators`  though technically speaking, with "parenthless" version, we could just merge it with `pyspark.sql.functions.udf`. The problem is it becomes rather ugly to document because the argument can be either `DataType` or `Callable`. 

This is why I wait with adding documentation but I pushed some doctests.